src/semant.ml

```diff
@@ -201,29 +201,50 @@ let rec check_expr expression table =
                    check_expr expr2 table, check_expr expr3 table))
     | TensorIdx(_,_) -> raise (Failure "Not yet implemented")
 
+(* If a function has a single type in its decl and the same
+ * id appears in its definition raise error, else return true*)
+let recursive_check (ftyp, fdef) =
+  let {types=ftyp'; _} = ftyp in
```

Comment:
What is this bracket thing? It also appears on line 223.

Reply:
Its a way to pattern match fields from a record. I found it here: https://www.cs.cornell.edu/courses/cs3110/2014sp/recitations/2/tuples_records_data.html

Reply:
Oh. I think that `let ftyp' = ftyp.types` would be a lot clearer, but if you're attached to this style that's fine.

Reply:
Not attached to it at all - I didn't know. I'll change it.

Reply:
Cool. Otherwise this looks good.